neuraxle/base.py

```diff
@@ -447,18 +447,74 @@ def setup(self, step_path: str, setup_arguments: dict = None) -> BaseStep:
         return self
 
     def set_hyperparams(self, hyperparams: HyperparameterSamples) -> BaseStep:
-        self.wrapped = self.wrapped.set_hyperparams(hyperparams.to_flat())
+        """
+        Set meta step and wrapped step hyperparams using the given hyperparams.
+
+        :param hyperparams: ordered dict containing all hyperparameters
+        :type hyperparams: HyperparameterSamples
+
+        :return:
+        """
+        hyperparams: HyperparameterSamples = HyperparameterSamples(hyperparams).to_nested_dict()
+
+        remainders = dict()
+        for name, hparams in hyperparams.items():
+            if name == self.wrapped.name:
+                self.wrapped.set_hyperparams(hparams)
+            else:
+                remainders[name] = hparams
+
+        self.hyperparams = HyperparameterSamples(remainders)
+
         return self
 
     def get_hyperparams(self) -> HyperparameterSamples:
-        return self.wrapped.get_hyperparams()
+        """
+        Get meta step and wrapped step hyperparams as a flat hyperparameter samples
+
+        :return: hyperparameters
+        """
+        return HyperparameterSamples(
+            dict(
+                self.hyperparams.to_flat_as_dict_primitive(),
+                **{self.wrapped.name: self.wrapped.hyperparams.to_flat_as_dict_primitive()}
+            )
+        ).to_flat()
```

Comment:
Don't we add strings at some point to do like `self.name + "__" + self.wrapped.name`? I'd think so (correct me if I'm wrong). Anyways, it should be similar to how the TruncableSteps works. 

Note: if so, "__" is a constant somewhere to be imported and used.

Reply:
.to_flat_as_dict_primitive() ends up calling a method that does that : 
```
def nested_dict_to_flat(nested_hyperparams, dict_ctor=OrderedDict):
    """
    Convert a nested hyperparameter dictionary to a flat one.

    :param nested_hyperparams: a nested hyperparameter dictionary.
    :param dict_ctor: ``OrderedDict`` by default. Will use this as a class to create the new returned dict.
    :return: a flat hyperparameter dictionary.
    """
    ret = dict_ctor()
    for k, v in nested_hyperparams.items():
        if isinstance(v, dict) or isinstance(v, OrderedDict) or isinstance(v, dict_ctor):
            _ret = nested_dict_to_flat(v)
            for key, val in _ret.items():
                ret[k + PARAMS_SPLIT_SEQ + key] = val
        else:
            ret[k] = v
    return ret
```

Reply:
you can see in the unit tests that this is tested. resolving this comment.